File: 프로그래머스/2/87946. 피로도/피로도.py
def solution(k, dungeons):
    answer = -1
    def dfs(visit, remain):
        res = 1
        for i in range(len(dungeons)):
            if visit[i] :
                continue
            if remain >= dungeons[i][0]:
                newVisit = list(visit)
                newVisit[i] = True
                currentCount = dfs(newVisit, remain - dungeons[i][1]) + 1
                res = currentCount if currentCount > res else res
        return res
    maxCount = 0
    for i in range(len(dungeons)):
        visit = [False for i in range(len(dungeons))]
        visit[i] = True
        count = dfs(visit, k - dungeons[i][1]);
        maxCount = maxCount if maxCount > count else count
    return maxCount

File: 프로그래머스/2/87946. 피로도/피로도.java
import java.util.*;
import java.io.*;
import java.util.stream.*;
class Solution {
    int answer = 0;
    public int solution(int k, int[][] dungeons) {
        for(int i = 0; i<dungeons.length; i++){
            boolean[] visit = new boolean[dungeons.length];
            if(dungeons[i][0] > k){
                continue;
            }
            visit[i] = true;
            dfs(k - dungeons[i][1], dungeons, visit, 1);
        }
        return answer;
    }
    public void dfs(int k, int[][] dungeons, boolean[] visit, int count) {
        for(int i = 0; i<dungeons.length; i++){
            if(visit[i]){
                continue;
            }
            if(dungeons[i][0] > k){
                continue;
            }
            visit[i] = true;
            dfs(k - dungeons[i][1], dungeons, visit, count + 1);
            visit[i] = false;
        }
        answer = Math.max(count, answer);
    }
}